### PR TITLE
feat: deploy with cloudbuild

### PR DESCRIPTION
**Pull request by caetano-colin**
I deployed the codebase using cloudbuild while documenting the necessary steps, main modifications are:

* README updates
* Docker image for the pipeline terraform runner (tf apply/plan) being created at 1-bootstrap
* Cloud Build Service Account permissions

Changes made up to only 4-applicationfactory,

### CODE REVIEW

File: 1-bootstrap/Dockerfile [hunk 1]
```diff
@@ -0,0 +1,39 @@
+# Copyright 2022 Google LLC
```
Comment (apeabody) [inline 1-bootstrap/Dockerfile:1]: Should this be?

```suggestion
# Copyright 2022-2024 Google LLC
```

File: 1-bootstrap/terraform.example.tfvars [hunk 2]
```diff
@@ -1 +1,53 @@
...
+    "network_self_link"  = "https://www.googleapis.com/compute/v1/projects/{PROJECT}/global/networks/{NETWORK}}"
+    "org_id"             = "{ORGANIZATION}"
+    "subnets_self_links" = [
+      "https://www.googleapis.com/compute/v1/projects/{PROJECT}/regions/{REGION}/subnetworks/{SUBNETWORK}",
+      "https://www.googleapis.com/compute/v1/projects/{PROJECT}/regions/{REGION}/subnetworks/{SUBNETWORK}",
+    ]
+  }
+}
+
+project_id = "REPLACE_WITH_YOUR_PROJECT"
```
Comment (apeabody) [inline 1-bootstrap/terraform.example.tfvars:53]:
```suggestion
project_id = "REPLACE_WITH_YOUR_PROJECT"

```

File: 1-bootstrap/outputs.tf [hunk 3]
```diff
@@ -38,3 +38,8 @@ output "source_repo_urls" {
   description = "Source repository URLs"
   value       = { for repo_id, repo in google_sourcerepo_repository.gcp_repo : repo_id => "https://source.developers.google.com/p/${var.project_id}/r/${repo.name}" }
 }
+
+output "cb_service_accounts_emails" {
+  description = "Service Accounts for the Multitenant Administration Cloud Build Triggers"
```
Comment (apeabody) [inline 1-bootstrap/outputs.tf:43]: Do we want to use `Multi-tenant` or `Multitenant`?  Currently there is a mix.

File: 1-bootstrap/Dockerfile [hunk 4]
```diff
@@ -0,0 +1,39 @@
...
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+FROM gcr.io/cloud-builders/gcloud-slim
+
+# Use ARG so that values can be overriden by user/cloudbuild
+ARG TERRAFORM_VERSION=1.5.7
```
Comment (apeabody) [inline 1-bootstrap/Dockerfile:18]: Is this only used for bootstrap?  FYI: fleetscope needs v1.6

**Review by apeabody — COMMENTED** [review]
Thanks @caetano-colin!

A few minor questions/comments

File: 1-bootstrap/outputs.tf (hunk 3 above)
Comment (gtsorbo) [reply 1-bootstrap/outputs.tf:43]: Let's stick with `multitenant`

File: 1-bootstrap/Dockerfile (hunk 4 above)
Comment (caetano-colin) [reply 1-bootstrap/Dockerfile:18]: This image would be used in all pipelines

do we want the latest terraform version? (v 1.9.6)

**Review by amandakarina — APPROVED** [review]

File: 1-bootstrap/Dockerfile (hunk 4 above)
Comment (apeabody) [reply 1-bootstrap/Dockerfile]: From call, let's use 1.6.6

**Review by gtsorbo — APPROVED** [review]

**Pull request merged**